Update how it works image and explanation (#592)
okteto/okteto: docs/how-does-it-work.md

## How does Okteto work?

> Okteto swaps your applications pods by a development environment:

Reviewer: ```suggestion
Okteto swaps your applications pod with a preconfigured development environment:
```

<img align="left" src="okteto-architecture.png">

> The **api-dev** pod duplicates the manifest of the **api** pod, so it is fully integrated with your Kubernetes setup.

Reviewer: this is not obvious a lot. I think you should mention how it's fully integrated.
